### ISSUE TITLE
[keymgr] Keymgr does not wait for OTP key valid bit

### ISSUE THREAD

**Issue by ballifatih** [lowRISC/opentitan#17521]
### Description

This is a spin-off from [the comment](https://github.com/lowRISC/opentitan/pull/17509#discussion_r1134678678).

The current `keymgr` implementation assumes that, when its internal FSM reaches to the stage where OTP root key is consumed, this key must be valid. In other words, `keymgr` does not wait for OTP root key to become valid, and proceeds to load an invalid key.

https://github.com/lowRISC/opentitan/blob/5e125d7908f156b07034b55806c992f7387e73b5/hw/ip/keymgr/rtl/keymgr_ctrl.sv#L336-L354

**Comment by ballifatih** [comment]
It is crucial to check that OTP root key becomes valid before keymgr FSM tries to latch the key. If we cannot guarantee that, this might be brick the chip, because keymgr will move to error state after boot.

As far as I understand, FSM inside `keymgr_ctrl` does not reach to latching state `StCtrlRootKey` until a first advance operation to keymgr is executed. But I do not know where this (i.e. transition from `Reset` to `Initialized`) happens for the first time. Is it ROM or ROMExt? @alphan @jadephilipoom @bilgiday 

Do you think existing top-level tests are sufficient to catch if PR #17509 leads to this scenario? Or do you think a SVA would ensure we do not get this sequence of events wrong? @cindychip 

cc: @johannheyszl @gdessouky

**Comment by jadephilipoom** [comment]
The testplan says the key manager should still be in the `Reset` state after ROM:
https://github.com/lowRISC/opentitan/blob/099596b1402ce0df8a980e936a1700dd287afe8a/sw/device/silicon_creator/rom/data/rom_e2e_testplan.hjson#L1173-L1193

So it seems safe to assume that this transition to `Initialized` happens in `ROM_EXT`.

**Comment by ballifatih** [comment]
Thanks @jadephilipoom. Good to know the first advance call is left to ROMExt.

Also as @bilgiday pointed out during offline discussion, in the power-up sequence, ROM execution comes strictly later after OTP is done. This can be seen in [fast domain FSM of pwrmgr](https://opentitan.org/book/hw/ip/pwrmgr/index.html#overall-sequencing), and in particular in the transition `init_otp` and `init_lc` which requires `otp_done`, and ROM does not start before `ack_powerup`.

To sum it up, the following sequence happens: OTP init is done -> ROM -starts > ROMExt starts -> first advance to keymgr

With this information in mind, once PR #17509 is merged and there are no nightly warning signs, it seems safe to close this issue.

**Comment by ballifatih** [comment]
Agreed with @cindychip, we can test this behavior at block-level by providing invalid OTP key when keymgr is sensing, and check if an error is raised by keymgr. I also think this is low priority.

**Comment by ballifatih** [comment]
> estimate 8

**Comment by ballifatih** [comment]
I think we can close this issue soon, but I will just wait for the TLT breakage to be fixed https://github.com/lowRISC/opentitan/issues/17834.

**Comment by ballifatih** [comment]
This issue was discussed in multiple occasions and the conclusion was that the power-up sequence already guarantees that OTP root key becomes valid before keymgr latches it. Therefore I am closing this issue.

I reverted the assertion `RootKeyValidDuringLatching_A` at #17945, as it was causing too much breakage in multiple top-level tests (when life cycle is not in PROD). LMK if you think closing this issue does not make sense @moidx @msfschaffner.

**Comment by msfschaffner** [comment]
I think this is ok to close.
